lib/src/settings/settings_controller.dart

```diff
@@ -7,41 +8,36 @@ import 'settings_service.dart';
 ///
 /// Controllers glue Data Services to Flutter Widgets. The SettingsController
 /// uses the SettingsService to store and retrieve user settings.
-class SettingsController with ChangeNotifier {
+class SettingsController {
   SettingsController(this._settingsService);
 
   // Make SettingsService a private variable so it is not used directly.
   final SettingsService _settingsService;
 
   // Make ThemeMode a private variable so it is not updated directly without
   // also persisting the changes with the SettingsService.
-  late ThemeMode _themeMode;
+  final _themeMode = signal(ThemeMode.system);
 
   // Allow Widgets to read the user's preferred ThemeMode.
-  ThemeMode get themeMode => _themeMode;
+  // Expose themeMode as a computed signal
+  Signal<ThemeMode> get themeMode => _themeMode;
```

Comment:
To restrict mutations to just `updateThemeMode`, would it make sense to type this as a `ReadonlySignal`?